optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +255,170 @@ public void skipAhead() throws ExecutionException, InterruptedException {
+                        consumedBestSolutions::add, null));
+            }
+            consumedSolutions.put(id, consumedBestSolutions);
+        }
+        assertInitializedJobs(jobs);
+        assertConsumedSolutions(problemCount, onlyFinalBestSolutions, consumedSolutions);
+    }
+
+    private void assertConsumedSolutions(int problemCount, boolean onlyFinalBestSolutions, Map<Integer, List<TestdataSolution>> consumedSolutions) {
+        if (onlyFinalBestSolutions) {
+            assertConsumedFinalBestSolutions(problemCount, consumedSolutions);
+        } else {
+            assertConsumedBestSolutions(problemCount, consumedSolutions);
+        }
+    }
+
+    private void assertConsumedFinalBestSolutions(int problemCount, Map<Integer, List<TestdataSolution>> consumedSolutions) {
+        for (int i = 0; i < problemCount; i++) {
+            assertThat(consumedSolutions.get(i)).hasSize(1);
+            TestdataSolution solution = consumedSolutions.get(i).get(0);
+            assertConsumedFinalBestSolution(solution);
+        }
+    }
+
+    private void assertConsumedFinalBestSolution(TestdataSolution solution) {
+        TestdataEntity entity = solution.getEntityList().get(0);
+        assertThat(entity.getCode()).isEqualTo("e1");
+        assertThat(entity.getValue().getCode()).isEqualTo("v1");
+        entity = solution.getEntityList().get(1);
+        assertThat(entity.getCode()).isEqualTo("e2");
+        assertThat(entity.getValue().getCode()).isEqualTo("v2");
+    }
+
+    private void assertConsumedBestSolutions(int problemCount, Map<Integer, List<TestdataSolution>> consumedSolutions) {
+        for (int i = 0; i < problemCount; i++) {
+            assertThat(consumedSolutions.get(i)).hasSize(2);
+            assertConsumedFirstBestSolution(consumedSolutions.get(i).get(0));
+            assertConsumedFinalBestSolution(consumedSolutions.get(i).get(1));
+        }
+    }
+
+    private void assertConsumedFirstBestSolution(TestdataSolution solution) {
+        TestdataEntity entity = solution.getEntityList().get(0);
+        assertThat(entity.getCode()).isEqualTo("e1");
+        assertThat(entity.getValue().getCode()).isEqualTo("v1");
+        entity = solution.getEntityList().get(1);
+        assertThat(entity.getCode()).isEqualTo("e2");
+        assertThat(entity.getValue()).isNull();
+    }
+
+    @Test(timeout = 1_000)
+    public void runSameIdProcesses_throwsIllegalStateException() {
+        SolverManagerConfig solverManagerConfig = new SolverManagerConfig();
+
+        SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataSolution.class, TestdataEntity.class)
+                .withPhases(createPhaseWithConcurrentSolvingStart(2));
+
+        SolverManager<TestdataSolution, Long> solverManager =
+                SolverManager.create(solverConfig, solverManagerConfig);
+
+        solverManager.solve(1L, PlannerTestUtils.generateTestdataSolution("solver"));
```

Comment:
using a solution code ID of "solver" is confusing. Please use "s1" which is short for solution 1.